Fix errors when building with an alternative libc than glibc

jaeyoonjung opened the pull request:
Otherwise it ends up failed when building with an alternative libc such as musl.

jaeyoonjung committed:
bluez5_%.bbappend: Add libc-glibc override for glibc-gconv-utf-*

Otherwise it ends up failed when building with an alternative libc such
as musl.

Signed-off-by: Jaeyoon Jung <jaeyoon.jung@lge.com>

lumag reviewed (changes requested):
This needs to go to OE-Core instead.

jaeyoonjung committed:
recipes: Remove glibc from DEPENDS

It's unnecessary and redundant as BASE_DEFAULT_DEPS already has
virtual/libc, and causing an error when building with an alternative
libc implementation such as musl.

Signed-off-by: Jaeyoon Jung <jaeyoon.jung@lge.com>

jaeyoonjung committed:
libdmabufheap: Add an optional TEMP_FAILURE_RETRY

Backports the commit 300f7b5c0d5e223441e4687113d55d4350859f9e from
upstream.

Signed-off-by: Jaeyoon Jung <jaeyoon.jung@lge.com>

jaeyoonjung committed:
cld80211-lib: Add TEMP_FAILURE_RETRY if undefined

Signed-off-by: Jaeyoon Jung <jaeyoon.jung@lge.com>

jaeyoonjung committed:
qcom-args: Fix errors when building with musl

Remove __unused macro and use __attribute__((unused)) instead.
Use -1 instead of NULL for SOCKET type.

Signed-off-by: Jaeyoon Jung <jaeyoon.jung@lge.com>

jaeyoonjung committed:
qcom-pa-bt-audio: Fix error when building with musl

Have HAVE_SYS_SOCKET_H defined by configure.

Signed-off-by: Jaeyoon Jung <jaeyoon.jung@lge.com>

jaeyoonjung committed:
qcom-pal: Fix errors when building with musl

Remove __unused macro and use __attribute__((unused)) instead.
'#include <unistd.h>' is needed by 'usleep' and 'close'.

Signed-off-by: Jaeyoon Jung <jaeyoon.jung@lge.com>

jaeyoonjung committed:
pa-pal-plugins: Fix error when building with musl

Add missing function declarations.
Have HAVE_SYS_SOCKET_H defined by configure.

Signed-off-by: Jaeyoon Jung <jaeyoon.jung@lge.com>

jaeyoonjung committed:
qcom-pa-pal-acd: Fix error when building with musl

Have HAVE_SYS_SOCKET_H defined by configure.

Signed-off-by: Jaeyoon Jung <jaeyoon.jung@lge.com>

jaeyoonjung committed:
qcom-pa-pal-voiceui: Fix error when building with musl

Remove __unused macro.
Use 0 over NULL for int type.

Signed-off-by: Jaeyoon Jung <jaeyoon.jung@lge.com>
